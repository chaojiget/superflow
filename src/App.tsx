import React, { useRef, useState } from 'react';
import FlowEditor from '@/components/FlowEditor';
import { copyText } from '@/utils';

const App: React.FC = () => {
  const inputRef = useRef<HTMLInputElement>(null);
  const [copyStatus, setCopyStatus] = useState('');

  const handleCopy = async () => {
    const text = inputRef.current?.value ?? '';
    const result = await copyText(text, inputRef.current || undefined);
    setCopyStatus(result.success ? '已复制' : result.message || '请手动复制');
  };

  return (
    <div>
      <FlowEditor />
      <div style={{ padding: 16 }}>
        <input ref={inputRef} placeholder="输入要复制的文本" />
        <button onClick={handleCopy} style={{ marginLeft: 8 }}>
          复制
        </button>
        {copyStatus && <p>{copyStatus}</p>}
      </div>
    </div>
  );
};

export default App;
