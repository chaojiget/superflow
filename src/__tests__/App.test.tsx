import { render, screen } from '@testing-library/react';
import App from '../App';

describe('App', () => {
  it('页面能够渲染', () => {
    render(<App />);
    expect(screen.getByText('Superflow 平台').outerHTML).toMatchSnapshot();
  });
});
