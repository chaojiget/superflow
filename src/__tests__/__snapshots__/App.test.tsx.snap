// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > 页面能够渲染 1`] = `"<button class="px-4 py-2 text-sm font-medium transition-all duration-200 cursor-pointer rounded-md bg-white text-blue-700 shadow-sm border border-blue-200">工作流编排 Studio</button>"`;
